jni/src/main/java/com/horizen/librustsidechains/FieldElement.java

```diff
@@ -12,9 +14,57 @@ private FieldElement(long fieldElementPointer) {
         this.fieldElementPointer = fieldElementPointer;
     }
 
-    private static native FieldElement nativeCreateFromLong(Long value);
+    private static native FieldElement nativeCreateFromLong(long value);
 
-    public static FieldElement createFromLong(Long value) {
+    public static FieldElement createFromLong(long value) {
         return nativeCreateFromLong(value);
     }
+
+    private static native FieldElement nativeCreateRandom();
+
+    public static FieldElement createRandom() { return nativeCreateRandom(); }
+
+    private native byte[] nativeSerializeFieldElement();
+
+    public byte[] serializeFieldElement() {
+        if (fieldElementPointer == 0)
+            throw new IllegalArgumentException("Field element was freed.");
+
+        return nativeSerializeFieldElement();
+    }
+
+    private static native long nativeDeserializeFieldElement(byte[] fieldElementBytes);
```

Comment:
Currently we cannot deserialize field element basing on random bytes. I think we need have method to deserialize (or create new) field element from random (unpredictible) bytes.

Reply:
Mmm is that really needed ? Are there situations in which you need to get a field element from random bytes ? If it's only for testing purposes, you have the function FieldElement.createRandom() that should be enough. Have you in mind specific use cases ?

Reply:
Resolved. As was discussed today on meeting.